src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1141,6 +1141,9 @@ private static bool PreferExpandedFormOverNormalForm<TMember>(MemberResolutionRe
                     }
 
                     break;
+
+                case MemberResolutionKind.None:
```

Comment:
>case MemberResolutionKind.None: [](http://example.com/codeflow?start=16&length=31)

I think it might be better to adjust the caller to never call this helper if ```IsMemberApplicableInNormalForm``` was not called. Basically, it is not normal to pass an uninitialized ```MemberResolutionResult``` around, The caller explicitly makes a choice to not initialize ```normalResult``` and, I think, it should be responsible for not passing it around. The logic could also be clearer because we will not have to rely on an indirect way to detect an uninitialized input. Also, it looks like, if ```Options.IsMethodGroupConversion``` is set, perhaps by mistake, the caller might still put the uninitialized resolution result into the output.